src/server/services/RedirectService.ts

```diff
@@ -50,6 +50,13 @@ export class RedirectService implements RedirectServiceInterface {
         logger.error(`Unable to increment click count: ${error}`),
       )
 
+    // Update click statistics for link.
+    this.urlRepository
+      .updateLinkStatistics(shortUrl, userAgent)
```

Comment:
can we merge `incrementClick` with `updateLinkStatistics` since they're really doing the same thing?

Reply:
it might be good to merge/migrate this functionality to the statistics service - we should try and minimise cross-communication between services and repositories.

Reply:
finally - may i suggest that `updateLinkStatistics` be moved to the link statistics service, since we rely on it for data retrieval as well? the idea being that we want services to talk to each other to get stuff done, instead of each one reaching into the repository directly.

Reply:
I have merged `incrementClick` into `updateLinkStatistics` and moved `updateLinkStatistics` into `LinkStatisticsService`.

Additionally, I shifted `incrementClick`, `updateDailyStatistics`, `updateWeekdayStatistics`, and `updateDeviceStatistics` from `UrlRepository` to `LinkStatisticsRepository` to centralise link statistics updates in one repository.